Archive/Examples/IfNormalization/WithoutAesop.lean

```diff
@@ -93,14 +93,22 @@ def normalize' (l : AList (fun _ : ℕ => Bool)) :
         · have := ht₃ v
           have := he₃ v
           simp_all? says simp_all only [Option.elim, normalized, Bool.and_eq_true,
-              Bool.not_eq_true', AList.lookup_insert, imp_false]
+              Bool.not_eq_true', implies_true]
           obtain ⟨⟨⟨tn, tc⟩, tr⟩, td⟩ := ht₂
           split <;> rename_i h'
           · subst h'
             simp_all
-          · simp_all? says simp_all only [hasNestedIf, Bool.or_self, hasConstantIf, and_self,
-              hasRedundantIf, Bool.or_false, beq_eq_false_iff_ne, ne_eq, not_false_eq_true,
-              disjoint, List.disjoint, decide_True, Bool.and_self]
+          · simp_all? says simp_all only [AList.lookup_insert_eq_none, ne_eq, hasNestedIf,
+              Bool.or_self, hasConstantIf, and_self, hasRedundantIf, Bool.or_false,
+              beq_eq_false_iff_ne, not_false_eq_true, disjoint, List.disjoint, decide_not,
+              Bool.and_true, Bool.and_eq_true, Bool.not_eq_true', decide_eq_false_iff_not, true_and]
+            constructor
+            · intro h
+              specialize ht₃ _ h
+              simp at ht₃
+            · intro h
+              specialize he₃ _ h
+              simp at he₃
```

Comment:
This is not great, but at least the aesop showcase in Result.lean has no regression now.

I'm content with this to go in as is, but if anyone feels like polishing of course go ahead.